var scores = [{
    name: 'Alice',
    start:0,
    end:160,
    boxes: [
      {start: 10,end:20},
      {start: 60,end:80},
      {start: 90,end:92}
    ]
  },
  {
    name: 'Alice',
    start:10,
    end:160,
    boxes: [
      {start: 10,end:20},
      {start: 120,end:160}
    ]
  },
  {
    name: 'Alice',
    start:10,
    end:160,
    boxes: [
      {start: 10,end:20},
      {start: 120,end:160}
    ]
  },
  {
    name: 'Alice',
    start:60,
    end:200,
    boxes: [
      {start: 70,end:80},
      {start: 120,end:160},
      {start: 170,end:190},
      {start: 192,end:200}

    ]
  },

];

var gene = {
  source: "ensembl_havana",
  object_type: "Gene",
  logic_name: "ensembl_havana_gene",
  seq_region_name: "1",
  db_type: "core",
  strand: -1,
  id: "ENSG00000134243",
  Transcript: [
  {
  source: "ensembl_havana",
  object_type: "Transcript",
  logic_name: "ensembl_havana_transcript",
  Exon: [
  {
  object_type: "Exon",
  version: 6,
  species: "homo_sapiens",
  assembly_name: "GRCh38",
  end: 109397951,
  seq_region_name: "1",
  db_type: "core",
  strand: -1,
  id: "ENSE00000958074",
  start: 109397587
  },
  {
  object_type: "Exon",
  version: 1,
  species: "homo_sapiens",
  assembly_name: "GRCh38",
  end: 109369589,
  seq_region_name: "1",
  db_type: "core",
  strand: -1,
  id: "ENSE00003493181",
  start: 109369530
  },
  {
  object_type: "Exon",
  version: 1,
  species: "homo_sapiens",
  assembly_name: "GRCh38",
  end: 109367481,
  seq_region_name: "1",
  db_type: "core",
  strand: -1,
  id: "ENSE00003483854",
  start: 109367408
  },
  {
  object_type: "Exon",
  version: 1,
  species: "homo_sapiens",
  assembly_name: "GRCh38",
  end: 109355469,
  seq_region_name: "1",
  db_type: "core",
  strand: -1,
  id: "ENSE00003496428",
  start: 109355367
  },
  {
  object_type: "Exon",
  version: 1,
  species: "homo_sapiens",
  assembly_name: "GRCh38",
  end: 109354531,
  seq_region_name: "1",
  db_type: "core",
  strand: -1,
  id: "ENSE00003682938",
  start: 109354367
  },
  {
  object_type: "Exon",
  version: 1,
  species: "homo_sapiens",
  assembly_name: "GRCh38",
  end: 109351002,
  seq_region_name: "1",
  db_type: "core",
  strand: -1,
  id: "ENSE00000912895",
  start: 109350929
  },
  {
  object_type: "Exon",
  version: 1,
  species: "homo_sapiens",
  assembly_name: "GRCh38",
  end: 109347532,
  seq_region_name: "1",
  db_type: "core",
  strand: -1,
  id: "ENSE00000912897",
  start: 109347483
  },
  {
  object_type: "Exon",
  version: 1,
  species: "homo_sapiens",
  assembly_name: "GRCh38",
  end: 109345881,
  seq_region_name: "1",
  db_type: "core",
  strand: -1,
  id: "ENSE00000912899",
  start: 109345751
  },
  {
  object_type: "Exon",
  version: 1,
  species: "homo_sapiens",
  assembly_name: "GRCh38",
  end: 109342158,
  seq_region_name: "1",
  db_type: "core",
  strand: -1,
  id: "ENSE00000912900",
  start: 109342014
  },
  {
  object_type: "Exon",
  version: 1,
  species: "homo_sapiens",
  assembly_name: "GRCh38",
  end: 109340879,
  seq_region_name: "1",
  db_type: "core",
  strand: -1,
  id: "ENSE00000912902",
  start: 109340724
  },
  {
  object_type: "Exon",
  version: 1,
  species: "homo_sapiens",
  assembly_name: "GRCh38",
  end: 109336346,
  seq_region_name: "1",
  db_type: "core",
  strand: -1,
  id: "ENSE00000912904",
  start: 109336240
  },
  {
  object_type: "Exon",
  version: 1,
  species: "homo_sapiens",
  assembly_name: "GRCh38",
  end: 109327601,
  seq_region_name: "1",
  db_type: "core",
  strand: -1,
  id: "ENSE00003629480",
  start: 109327499
  },
  {
  object_type: "Exon",
  version: 1,
  species: "homo_sapiens",
  assembly_name: "GRCh38",
  end: 109327160,
  seq_region_name: "1",
  db_type: "core",
  strand: -1,
  id: "ENSE00000912908",
  start: 109326992
  },
  {
  object_type: "Exon",
  version: 1,
  species: "homo_sapiens",
  assembly_name: "GRCh38",
  end: 109325089,
  seq_region_name: "1",
  db_type: "core",
  strand: -1,
  id: "ENSE00001152449",
  start: 109324899
  },
  {
  object_type: "Exon",
  version: 1,
  species: "homo_sapiens",
  assembly_name: "GRCh38",
  end: 109323121,
  seq_region_name: "1",
  db_type: "core",
  strand: -1,
  id: "ENSE00000912912",
  start: 109322932
  },
  {
  object_type: "Exon",
  version: 1,
  species: "homo_sapiens",
  assembly_name: "GRCh38",
  end: 109317969,
  seq_region_name: "1",
  db_type: "core",
  strand: -1,
  id: "ENSE00000912914",
  start: 109317853
  },
  {
  object_type: "Exon",
  version: 1,
  species: "homo_sapiens",
  assembly_name: "GRCh38",
  end: 109316958,
  seq_region_name: "1",
  db_type: "core",
  strand: -1,
  id: "ENSE00001152424",
  start: 109316850
  },
  {
  object_type: "Exon",
  version: 1,
  species: "homo_sapiens",
  assembly_name: "GRCh38",
  end: 109314778,
  seq_region_name: "1",
  db_type: "core",
  strand: -1,
  id: "ENSE00003483084",
  start: 109314672
  },
  {
  object_type: "Exon",
  version: 1,
  species: "homo_sapiens",
  assembly_name: "GRCh38",
  end: 109314384,
  seq_region_name: "1",
  db_type: "core",
  strand: -1,
  id: "ENSE00001124417",
  start: 109314261
  },
  {
  object_type: "Exon",
  version: 2,
  species: "homo_sapiens",
  assembly_name: "GRCh38",
  end: 109314057,
  seq_region_name: "1",
  db_type: "core",
  strand: -1,
  id: "ENSE00001348032",
  start: 109309570
  }
  ],
  Parent: "ENSG00000134243",
  seq_region_name: "1",
  db_type: "core",
  is_canonical: 1,
  strand: -1,
  id: "ENST00000256637",
  version: 7,
  species: "homo_sapiens",
  assembly_name: "GRCh38",
  display_name: "SORT1-201",
  end: 109397951,
  biotype: "protein_coding",
  Translation: {
  object_type: "Translation",
  species: "homo_sapiens",
  Parent: "ENST00000256637",
  end: 109397892,
  length: 831,
  db_type: "core",
  id: "ENSP00000256637",
  start: 109314043
  },
  start: 109309570
  },
  {
  source: "havana",
  object_type: "Transcript",
  logic_name: "havana",
  Exon: [
  {
  object_type: "Exon",
  version: 1,
  species: "homo_sapiens",
  assembly_name: "GRCh38",
  end: 109316927,
  seq_region_name: "1",
  db_type: "core",
  strand: -1,
  id: "ENSE00001878523",
  start: 109316850
  },
  {
  object_type: "Exon",
  version: 1,
  species: "homo_sapiens",
  assembly_name: "GRCh38",
  end: 109314778,
  seq_region_name: "1",
  db_type: "core",
  strand: -1,
  id: "ENSE00003628909",
  start: 109314672
  },
  {
  object_type: "Exon",
  version: 1,
  species: "homo_sapiens",
  assembly_name: "GRCh38",
  end: 109314057,
  seq_region_name: "1",
  db_type: "core",
  strand: -1,
  id: "ENSE00001840440",
  start: 109313801
  }
  ],
  Parent: "ENSG00000134243",
  seq_region_name: "1",
  db_type: "core",
  is_canonical: 0,
  strand: -1,
  id: "ENST00000485149",
  version: 1,
  species: "homo_sapiens",
  assembly_name: "GRCh38",
  display_name: "SORT1-206",
  end: 109316927,
  biotype: "processed_transcript",
  start: 109313801
  },
  {
  source: "havana",
  object_type: "Transcript",
  logic_name: "havana",
  Exon: [
  {
  object_type: "Exon",
  version: 1,
  species: "homo_sapiens",
  assembly_name: "GRCh38",
  end: 109336302,
  seq_region_name: "1",
  db_type: "core",
  strand: -1,
  id: "ENSE00001866815",
  start: 109336240
  },
  {
  object_type: "Exon",
  version: 1,
  species: "homo_sapiens",
  assembly_name: "GRCh38",
  end: 109327601,
  seq_region_name: "1",
  db_type: "core",
  strand: -1,
  id: "ENSE00003520573",
  start: 109327499
  },
  {
  object_type: "Exon",
  version: 1,
  species: "homo_sapiens",
  assembly_name: "GRCh38",
  end: 109327160,
  seq_region_name: "1",
  db_type: "core",
  strand: -1,
  id: "ENSE00001922227",
  start: 109326806
  }
  ],
  Parent: "ENSG00000134243",
  seq_region_name: "1",
  db_type: "core",
  is_canonical: 0,
  strand: -1,
  id: "ENST00000466471",
  version: 1,
  species: "homo_sapiens",
  assembly_name: "GRCh38",
  display_name: "SORT1-202",
  end: 109336302,
  biotype: "retained_intron",
  start: 109326806
  },
  {
  source: "havana",
  object_type: "Transcript",
  logic_name: "havana",
  Exon: [
  {
  object_type: "Exon",
  version: 1,
  species: "homo_sapiens",
  assembly_name: "GRCh38",
  end: 109389899,
  seq_region_name: "1",
  db_type: "core",
  strand: -1,
  id: "ENSE00001839735",
  start: 109389775
  },
  {
  object_type: "Exon",
  version: 1,
  species: "homo_sapiens",
  assembly_name: "GRCh38",
  end: 109369589,
  seq_region_name: "1",
  db_type: "core",
  strand: -1,
  id: "ENSE00003675179",
  start: 109369530
  },
  {
  object_type: "Exon",
  version: 1,
  species: "homo_sapiens",
  assembly_name: "GRCh38",
  end: 109367481,
  seq_region_name: "1",
  db_type: "core",
  strand: -1,
  id: "ENSE00003782720",
  start: 109367408
  },
  {
  object_type: "Exon",
  version: 1,
  species: "homo_sapiens",
  assembly_name: "GRCh38",
  end: 109355469,
  seq_region_name: "1",
  db_type: "core",
  strand: -1,
  id: "ENSE00003496428",
  start: 109355367
  },
  {
  object_type: "Exon",
  version: 1,
  species: "homo_sapiens",
  assembly_name: "GRCh38",
  end: 109354531,
  seq_region_name: "1",
  db_type: "core",
  strand: -1,
  id: "ENSE00003682938",
  start: 109354367
  },
  {
  object_type: "Exon",
  version: 1,
  species: "homo_sapiens",
  assembly_name: "GRCh38",
  end: 109351002,
  seq_region_name: "1",
  db_type: "core",
  strand: -1,
  id: "ENSE00001953566",
  start: 109350947
  }
  ],
  Parent: "ENSG00000134243",
  seq_region_name: "1",
  db_type: "core",
  is_canonical: 0,
  strand: -1,
  id: "ENST00000493736",
  version: 5,
  species: "homo_sapiens",
  assembly_name: "GRCh38",
  display_name: "SORT1-207",
  end: 109389899,
  biotype: "protein_coding",
  Translation: {
  object_type: "Translation",
  species: "homo_sapiens",
  Parent: "ENST00000493736",
  end: 109367439,
  length: 118,
  db_type: "core",
  id: "ENSP00000488262",
  start: 109350947
  },
  start: 109350947
  },
  {
  source: "havana",
  object_type: "Transcript",
  logic_name: "havana",
  Exon: [
  {
  object_type: "Exon",
  version: 1,
  species: "homo_sapiens",
  assembly_name: "GRCh38",
  end: 109393357,
  seq_region_name: "1",
  db_type: "core",
  strand: -1,
  id: "ENSE00001856884",
  start: 109393027
  },
  {
  object_type: "Exon",
  version: 1,
  species: "homo_sapiens",
  assembly_name: "GRCh38",
  end: 109369589,
  seq_region_name: "1",
  db_type: "core",
  strand: -1,
  id: "ENSE00003675179",
  start: 109369530
  },
  {
  object_type: "Exon",
  version: 1,
  species: "homo_sapiens",
  assembly_name: "GRCh38",
  end: 109367481,
  seq_region_name: "1",
  db_type: "core",
  strand: -1,
  id: "ENSE00003782720",
  start: 109367408
  },
  {
  object_type: "Exon",
  version: 1,
  species: "homo_sapiens",
  assembly_name: "GRCh38",
  end: 109355469,
  seq_region_name: "1",
  db_type: "core",
  strand: -1,
  id: "ENSE00003496428",
  start: 109355367
  },
  {
  object_type: "Exon",
  version: 1,
  species: "homo_sapiens",
  assembly_name: "GRCh38",
  end: 109354531,
  seq_region_name: "1",
  db_type: "core",
  strand: -1,
  id: "ENSE00001898203",
  start: 109354485
  }
  ],
  Parent: "ENSG00000134243",
  seq_region_name: "1",
  db_type: "core",
  is_canonical: 0,
  strand: -1,
  id: "ENST00000482236",
  version: 5,
  species: "homo_sapiens",
  assembly_name: "GRCh38",
  display_name: "SORT1-204",
  end: 109393357,
  biotype: "protein_coding",
  Translation: {
  object_type: "Translation",
  species: "homo_sapiens",
  Parent: "ENST00000482236",
  end: 109367439,
  length: 60,
  db_type: "core",
  id: "ENSP00000487712",
  start: 109354485
  },
  start: 109354485
  },
  {
  source: "havana",
  object_type: "Transcript",
  logic_name: "havana",
  Exon: [
  {
  object_type: "Exon",
  version: 1,
  species: "homo_sapiens",
  assembly_name: "GRCh38",
  end: 109392988,
  seq_region_name: "1",
  db_type: "core",
  strand: -1,
  id: "ENSE00003782073",
  start: 109392591
  },
  {
  object_type: "Exon",
  version: 1,
  species: "homo_sapiens",
  assembly_name: "GRCh38",
  end: 109369589,
  seq_region_name: "1",
  db_type: "core",
  strand: -1,
  id: "ENSE00003675179",
  start: 109369530
  },
  {
  object_type: "Exon",
  version: 1,
  species: "homo_sapiens",
  assembly_name: "GRCh38",
  end: 109367481,
  seq_region_name: "1",
  db_type: "core",
  strand: -1,
  id: "ENSE00003782720",
  start: 109367408
  },
  {
  object_type: "Exon",
  version: 2,
  species: "homo_sapiens",
  assembly_name: "GRCh38",
  end: 109355469,
  seq_region_name: "1",
  db_type: "core",
  strand: -1,
  id: "ENSE00003589952",
  start: 109355441
  }
  ],
  Parent: "ENSG00000134243",
  seq_region_name: "1",
  db_type: "core",
  is_canonical: 0,
  strand: -1,
  id: "ENST00000633956",
  version: 1,
  species: "homo_sapiens",
  assembly_name: "GRCh38",
  display_name: "SORT1-210",
  end: 109392988,
  biotype: "protein_coding",
  Translation: {
  object_type: "Translation",
  species: "homo_sapiens",
  Parent: "ENST00000633956",
  end: 109367439,
  length: 20,
  db_type: "core",
  id: "ENSP00000488662",
  start: 109355441
  },
  start: 109355441
  },
  {
  source: "havana",
  object_type: "Transcript",
  logic_name: "havana",
  Exon: [
  {
  object_type: "Exon",
  version: 1,
  species: "homo_sapiens",
  assembly_name: "GRCh38",
  end: 109369589,
  seq_region_name: "1",
  db_type: "core",
  strand: -1,
  id: "ENSE00003675179",
  start: 109369530
  },
  {
  object_type: "Exon",
  version: 1,
  species: "homo_sapiens",
  assembly_name: "GRCh38",
  end: 109367481,
  seq_region_name: "1",
  db_type: "core",
  strand: -1,
  id: "ENSE00001919816",
  start: 109366783
  }
  ],
  Parent: "ENSG00000134243",
  seq_region_name: "1",
  db_type: "core",
  is_canonical: 0,
  strand: -1,
  id: "ENST00000483508",
  version: 5,
  species: "homo_sapiens",
  assembly_name: "GRCh38",
  display_name: "SORT1-205",
  end: 109369589,
  biotype: "retained_intron",
  start: 109366783
  },
  {
  source: "havana",
  object_type: "Transcript",
  logic_name: "havana",
  Exon: [
  {
  object_type: "Exon",
  version: 1,
  species: "homo_sapiens",
  assembly_name: "GRCh38",
  end: 109369589,
  seq_region_name: "1",
  db_type: "core",
  strand: -1,
  id: "ENSE00003675179",
  start: 109369530
  },
  {
  object_type: "Exon",
  version: 1,
  species: "homo_sapiens",
  assembly_name: "GRCh38",
  end: 109368808,
  seq_region_name: "1",
  db_type: "core",
  strand: -1,
  id: "ENSE00001931600",
  start: 109368478
  }
  ],
  Parent: "ENSG00000134243",
  seq_region_name: "1",
  db_type: "core",
  is_canonical: 0,
  strand: -1,
  id: "ENST00000495777",
  version: 5,
  species: "homo_sapiens",
  assembly_name: "GRCh38",
  display_name: "SORT1-208",
  end: 109369589,
  biotype: "processed_transcript",
  start: 109368478
  },
  {
  source: "havana",
  object_type: "Transcript",
  logic_name: "havana",
  Exon: [
  {
  object_type: "Exon",
  version: 1,
  species: "homo_sapiens",
  assembly_name: "GRCh38",
  end: 109397861,
  seq_region_name: "1",
  db_type: "core",
  strand: -1,
  id: "ENSE00001937807",
  start: 109397587
  },
  {
  object_type: "Exon",
  version: 1,
  species: "homo_sapiens",
  assembly_name: "GRCh38",
  end: 109397340,
  seq_region_name: "1",
  db_type: "core",
  strand: -1,
  id: "ENSE00001905622",
  start: 109397207
  },
  {
  object_type: "Exon",
  version: 1,
  species: "homo_sapiens",
  assembly_name: "GRCh38",
  end: 109369589,
  seq_region_name: "1",
  db_type: "core",
  strand: -1,
  id: "ENSE00003675179",
  start: 109369530
  }
  ],
  Parent: "ENSG00000134243",
  seq_region_name: "1",
  db_type: "core",
  is_canonical: 0,
  strand: -1,
  id: "ENST00000471996",
  version: 1,
  species: "homo_sapiens",
  assembly_name: "GRCh38",
  display_name: "SORT1-203",
  end: 109397861,
  biotype: "nonsense_mediated_decay",
  Translation: {
  object_type: "Translation",
  species: "homo_sapiens",
  Parent: "ENST00000471996",
  end: 109397861,
  length: 104,
  db_type: "core",
  id: "ENSP00000487674",
  start: 109397302
  },
  start: 109369530
  },
  {
  source: "ensembl",
  object_type: "Transcript",
  logic_name: "ensembl",
  Exon: [
  {
  object_type: "Exon",
  version: 1,
  species: "homo_sapiens",
  assembly_name: "GRCh38",
  end: 109393357,
  seq_region_name: "1",
  db_type: "core",
  strand: -1,
  id: "ENSE00001856884",
  start: 109393027
  },
  {
  object_type: "Exon",
  version: 1,
  species: "homo_sapiens",
  assembly_name: "GRCh38",
  end: 109369589,
  seq_region_name: "1",
  db_type: "core",
  strand: -1,
  id: "ENSE00003675179",
  start: 109369530
  },
  {
  object_type: "Exon",
  version: 1,
  species: "homo_sapiens",
  assembly_name: "GRCh38",
  end: 109367481,
  seq_region_name: "1",
  db_type: "core",
  strand: -1,
  id: "ENSE00003782720",
  start: 109367408
  },
  {
  object_type: "Exon",
  version: 1,
  species: "homo_sapiens",
  assembly_name: "GRCh38",
  end: 109355469,
  seq_region_name: "1",
  db_type: "core",
  strand: -1,
  id: "ENSE00003496428",
  start: 109355367
  },
  {
  object_type: "Exon",
  version: 1,
  species: "homo_sapiens",
  assembly_name: "GRCh38",
  end: 109354531,
  seq_region_name: "1",
  db_type: "core",
  strand: -1,
  id: "ENSE00003682938",
  start: 109354367
  },
  {
  object_type: "Exon",
  version: 1,
  species: "homo_sapiens",
  assembly_name: "GRCh38",
  end: 109351002,
  seq_region_name: "1",
  db_type: "core",
  strand: -1,
  id: "ENSE00000912895",
  start: 109350929
  },
  {
  object_type: "Exon",
  version: 1,
  species: "homo_sapiens",
  assembly_name: "GRCh38",
  end: 109347532,
  seq_region_name: "1",
  db_type: "core",
  strand: -1,
  id: "ENSE00000912897",
  start: 109347483
  },
  {
  object_type: "Exon",
  version: 1,
  species: "homo_sapiens",
  assembly_name: "GRCh38",
  end: 109345878,
  seq_region_name: "1",
  db_type: "core",
  strand: -1,
  id: "ENSE00002220355",
  start: 109345751
  },
  {
  object_type: "Exon",
  version: 1,
  species: "homo_sapiens",
  assembly_name: "GRCh38",
  end: 109342158,
  seq_region_name: "1",
  db_type: "core",
  strand: -1,
  id: "ENSE00000912900",
  start: 109342014
  },
  {
  object_type: "Exon",
  version: 1,
  species: "homo_sapiens",
  assembly_name: "GRCh38",
  end: 109340879,
  seq_region_name: "1",
  db_type: "core",
  strand: -1,
  id: "ENSE00000912902",
  start: 109340724
  },
  {
  object_type: "Exon",
  version: 1,
  species: "homo_sapiens",
  assembly_name: "GRCh38",
  end: 109336346,
  seq_region_name: "1",
  db_type: "core",
  strand: -1,
  id: "ENSE00000912904",
  start: 109336240
  },
  {
  object_type: "Exon",
  version: 1,
  species: "homo_sapiens",
  assembly_name: "GRCh38",
  end: 109327601,
  seq_region_name: "1",
  db_type: "core",
  strand: -1,
  id: "ENSE00003629480",
  start: 109327499
  },
  {
  object_type: "Exon",
  version: 1,
  species: "homo_sapiens",
  assembly_name: "GRCh38",
  end: 109327160,
  seq_region_name: "1",
  db_type: "core",
  strand: -1,
  id: "ENSE00000912908",
  start: 109326992
  },
  {
  object_type: "Exon",
  version: 1,
  species: "homo_sapiens",
  assembly_name: "GRCh38",
  end: 109325089,
  seq_region_name: "1",
  db_type: "core",
  strand: -1,
  id: "ENSE00001152449",
  start: 109324899
  },
  {
  object_type: "Exon",
  version: 1,
  species: "homo_sapiens",
  assembly_name: "GRCh38",
  end: 109323121,
  seq_region_name: "1",
  db_type: "core",
  strand: -1,
  id: "ENSE00000912912",
  start: 109322932
  },
  {
  object_type: "Exon",
  version: 1,
  species: "homo_sapiens",
  assembly_name: "GRCh38",
  end: 109317969,
  seq_region_name: "1",
  db_type: "core",
  strand: -1,
  id: "ENSE00000912914",
  start: 109317853
  },
  {
  object_type: "Exon",
  version: 1,
  species: "homo_sapiens",
  assembly_name: "GRCh38",
  end: 109316958,
  seq_region_name: "1",
  db_type: "core",
  strand: -1,
  id: "ENSE00001152424",
  start: 109316850
  },
  {
  object_type: "Exon",
  version: 1,
  species: "homo_sapiens",
  assembly_name: "GRCh38",
  end: 109314778,
  seq_region_name: "1",
  db_type: "core",
  strand: -1,
  id: "ENSE00003483084",
  start: 109314672
  },
  {
  object_type: "Exon",
  version: 1,
  species: "homo_sapiens",
  assembly_name: "GRCh38",
  end: 109314384,
  seq_region_name: "1",
  db_type: "core",
  strand: -1,
  id: "ENSE00001124417",
  start: 109314261
  },
  {
  object_type: "Exon",
  version: 2,
  species: "homo_sapiens",
  assembly_name: "GRCh38",
  end: 109314057,
  seq_region_name: "1",
  db_type: "core",
  strand: -1,
  id: "ENSE00002294602",
  start: 109309568
  }
  ],
  Parent: "ENSG00000134243",
  seq_region_name: "1",
  db_type: "core",
  is_canonical: 0,
  strand: -1,
  id: "ENST00000538502",
  version: 5,
  species: "homo_sapiens",
  assembly_name: "GRCh38",
  display_name: "SORT1-209",
  end: 109393357,
  biotype: "protein_coding",
  Translation: {
  object_type: "Translation",
  species: "homo_sapiens",
  Parent: "ENST00000538502",
  end: 109367439,
  length: 694,
  db_type: "core",
  id: "ENSP00000438597",
  start: 109314043
  },
  start: 109309568
  }
  ],
  version: 11,
  species: "homo_sapiens",
  assembly_name: "GRCh38",
  display_name: "SORT1",
  description: "sortilin 1 [Source:HGNC Symbol;Acc:HGNC:11186]",
  end: 109397951,
  biotype: "protein_coding",
  start: 109309568
}


var simpleGene = {
  start: gene.start,
  end: gene.end,
  transcripts: gene.Transcript.map(transcript => {
    return {
      id: transcript.id,
      start: transcript.start,
      end: transcript.end,
      exons: transcript.Exon.map(exon => ({
        start: exon.start,
        end: exon.end
      }))
    }
  })
}

console.log(simpleGene)

var margin = {top: 20, right: 0, bottom: 60, left: 140}
var width = 800 - margin.left - margin.right;
var height = 400 - margin.top - margin.bottom;

var yScale = d3.scaleBand()
  .paddingInner(.6)
  .domain(simpleGene.transcripts.map(d => d.id))
  .range([0, height]);

var xScale = d3.scaleLinear()
  .domain([simpleGene.start, simpleGene.end+ 10000])
  .range([0, width])

var yAxis = d3.axisLeft(yScale)
var xAxis = d3.axisBottom(xScale)

var svg = d3.select('svg')
    .attr("width", width + margin.left + margin.right)
    .attr("height", height + margin.top + margin.bottom)
    .append('g')
    .attr('transform', `translate(${margin.left}, ${margin.top})`)
  
var barContainer  = svg.selectAll('g')
  .data(simpleGene.transcripts)
  .enter()
    .append('g')
      .attr('transform', (d, i) => 'translate(30, ' + i * 33 + ')')

var geneBar = barContainer
    .append('rect')
      .attr('width', function(d) {return xScale(d.end)-xScale(d.start)} )
      .attr('height', 5)
      .attr('x', function(d) { return xScale(d.start)})  
      .attr('y', 4)
      .attr('class', 'bar')

var eachGene = barContainer
  .selectAll('rect')
  .data( d => d.exons)
  .enter()
  .append('rect')
    .attr('width', function(d) {return xScale(d.end)-xScale(d.start)} )
    .attr('height', 15)
    .attr('fill', 'orange')
    .attr('stroke-width', 1)
    .attr('stroke', 'black')
    .attr('x', function(d) { return xScale(d.start)})  

svg.append('g')
  .attr('transform', `translate(${0}, ${margin.top+ height})`)
  .call(xAxis)
  .selectAll('text')
  .style('text-anchor', 'end')
  .attr('transform', `rotate(-15)`)

svg.append('g')
  .call(yAxis)
    


// bar.append('rect')
//   .style('width', d => d.boxs.end)
//   .attr('x', d => d.boxs.start)
//   .attr('class', 'bar')

// bar.append('rect')
//   .style('width', d => d.score)
//   .attr('x', d => d.start)
//   .attr('class', 'bar')
//   .on('mouseover', function (d, i, elements) {
//     d3.select(this).style('transform', 'scaleX(2)');
//     d3.selectAll(elements)
//       .filter(':not(:hover)')
//       .style('fill-opacity', 0.5);
//   })
//   .on('mouseout', function (d, i, elements) {
//     d3.select(this).style('transform', 'scaleX(1)');
//     d3.selectAll(elements)
//       .style('fill-opacity', 1);
//   });



// bar.append('text')
//   .attr('y', 20)
//   .text(function (d) {
//     return d.name;
//   });

// 
var scores = [
  { name: 'Alice', score: 96 },
  { name: 'Billy', score: 83 },
  { name: 'Cindy', score: 91 },
  { name: 'David', score: 96 },
  { name: 'Emily', score: 88 }
];

var bar = d3.select('.ex1')
  .append('svg')
  .selectAll('rect')
  .data(scores)
  .enter()
    .append('g')
    .attr('transform', (d, i) => 'translate(0, ' + i * 33 + ')')

bar.append('rect')
  .style('width', d => d.score)
  .attr('class', 'bar')
  .on('mouseover', function(d,i,elements){
    d3.select(this)
      .call(scaleBar, 2)
      .call(setFill, 'orange')
    d3.selectAll(elements)
      .filter(':not(:hover)')
      .call(fade, 0.5)
  })
  .on('mouseout', function(d,i,elements){
    d3.select(this)
      .call(scaleBar, 1)
      .call(setFill, 'lightgreen')
    d3.selectAll(elements)
      .call(fade, 1)
  })

bar.append('text')
  .attr('y', 20)
  .text(function(d) {
    return d.name
})



function scaleBar(selection, scale){
  selection.style('transform', 'scaleX(' + scale + ')')
}

function setFill (selection, color){
  selection.style('fill', color)
}

function fade (selection, opacity){
  selection.style('fill-opacity', opacity)
}



var data = [
  {score: 63, subject: 'Mathematics'},
  {score: 82, subject: 'Geography'},
  {score: 74, subject: 'Spelling'},
  {score: 97, subject: 'Reading'},
  {score: 52, subject: 'Science'},
  {score: 74, subject: 'Chemistry'},
  {score: 97, subject: 'Physics'},
  {score: 52, subject: 'ASL'}
];

var margin = {top: 10, right: 20, bottom: 60, left:40}
var width = 500 - margin.left - margin.right;
var height = 700 - margin.top - margin.bottom; 

// var fullWidth = width + margin.left + margin.right
// var fullHeight = height + margin.top + margin.bottom

var svg = d3.select('.ex2')
  .append('svg')
    .attr('width', width + margin.left + margin.right)
    .attr('height', height + margin.top + margin.bottom)
    .call(responsivefy)
    // .attr('viewBox', `0 0 ${fullWidth/2} ${fullHeight/2}`)
  .append('g')
    .attr('transform', `translate(${margin.left}, ${margin.top})`)

// svg.append('rect')
//   .attr('width', width)
//   .attr('height', height)
//   .style('fill', 'lightblue')
//   .style('stroke', 'green')

var yScale = d3.scaleLinear()
    .domain([0, 100])
    .range([height, 0])
var yAxis = d3.axisLeft(yScale).ticks(5)
svg.call(yAxis)

var xScale = d3.scaleBand()
  .padding(0.2)
  .domain(data.map(d => d.subject))
  .range([0, width]);

var xAxis = d3.axisBottom(xScale).ticks(5)
svg.append('g')
  .attr('transform', `translate( 0, ${height})`)
  .call(xAxis)
  .selectAll('text')
  .style('text-anchor', 'end')
  .attr('transform', `rotate(-45)`)

svg.selectAll('rect')
  .data(data)
  .enter()
  .append('rect')
  .attr('x', d => xScale(d.subject))
  .attr('y', d => yScale(d.score))
  .attr('width', xScale.bandwidth())
  .attr('height', d=> height - yScale(d.score))
  .attr('fill', `steelblue`)
function responsivefy(svg) {
  // get container + svg aspect ratio
  var container = d3.select(svg.node().parentNode),
    width = parseInt(svg.style("width")),
    height = parseInt(svg.style("height")),
    aspect = width / height;
  
  // add viewBox and preserveAspectRatio properties,
  // and call resize so that svg resizes on inital page load
  svg.attr("viewBox", "0 0 " + width + " " + height)
    .attr("preserveAspectRatio", "xMinYMid")
    .call(resize);
  
  // to register multiple listeners for same event type,
  // you need to add namespace, i.e., 'click.foo'
  // necessary if you call invoke this function for multiple svgs
  // api docs: https://github.com/mbostock/d3/wiki/Selections#on
  d3.select(window).on("resize." + container.attr("id"), resize);
  
  // get width of container and resize svg to fit it
  function resize() {
    var targetWidth = parseInt(container.style("width"));
    svg.attr("width", targetWidth);
    svg.attr("height", Math.round(targetWidth / aspect));
  }
}


console.log(svg)